src/hooks/usePersistentQuery.ts

```diff
@@ -0,0 +1,20 @@
+import { useNavigate } from 'react-router-dom';
+import { getBasePath } from 'utils/formatPath';
+import useQueryParams from 'hooks/useQueryParams';
+import { usePersistentGlobalState } from './usePersistentGlobalState';
+
+export const usePersistentQuery = (key: string) => {
```

Comment:
Does this store the same state in both localStorage and the query string? If so, would one place be enough? 

Also seems unused.